### DISCUSSION TITLE
FileManager notification when upload is completed

### DISCUSSION THREAD

**Discussion by mikecasas** [oqtane/oqtane.framework#745] [Q&A]
I have a use case to process uploaded files. FileManager is a great component and has almost everything needed, but is there a way to know when a file/files are completed uploading, so then I can begin a secondary process?

**Comment by sbwalker** [comment]
Its not clear from your description if you are looking for a synchronous or asynchronous solution? I am assuming asynchronous - and if so, then perhaps a possibility is to create a scheduled job which monitors a folder every minute?

**Reply by mikecasas** [reply]
In my custom code, I modified FileManager to expose an event, then called it when the file was successfully uploaded.

**Reply by hishamco** [reply]
Please defer this after I push the alert component that suggested [here](https://github.com/oqtane/oqtane.framework/pull/747#issuecomment-694040813)

**Reply by sbwalker** [reply]
@mikecasas so is your solution synchronous or asynchronous. Can the event be hooked on any active client or only the client that uploaded the file?

**Reply by mikecasas** [reply]
I'm using it async. I believe you can use Blazor events either way??? I'm not sure about the second part of your question, but I assume it's only hooked for the person uploading the file.

In FileManager I added a property and inserted the call at the end of the UploadFile method.

```
    [Parameter]
    public EventCallback<bool> UploadCompleted { get; set; }


 if (upload.Length == 1)
                    {
                        var file = _files.Where(item => item.Name == upload[0]).FirstOrDefault();
                        if (file != null)
                        {
                            FileId = file.FileId;
                            await SetImage();
                        }
                    }
                    StateHasChanged();


                    await UploadCompleted.InvokeAsync(true);


```
Then in my custom module I am using it:


```
   <FileManager UploadMultiple="true" ShowFiles="false" FolderId="@_folderId" Folder="@customFolder" UploadCompleted="@(async () => await UploadCompleted())" />


    private async Task UploadCompleted()
    {
        await ProcessFile();
    }
    private async Task ProcessFile()
    {
          // custom logic here
    }

```
